hoomd/mpcd/ConcentricCylindersGeometry.h

```diff
@@ -0,0 +1,193 @@
+        // exit immediately if no collision is found or particle is stationary in xy plane
+        const Scalar vxy_sq = vel.x * vel.x + vel.y * vel.y;
+        if (sign == 0 || vxy_sq == Scalar(0))
+            {
+            dt = Scalar(0);
+            return false;
+            }
+
+        /*
+         * Find the time remaining when the particle is collided with wall. This time is computed
+         * by backtracking the position, dot(pos, pos) = R^2.
+         */
+        const Scalar Rsq = (sign == 1) ? m_R0_sq : m_R1_sq;
+        const Scalar b = -Scalar(2) * (pos.x * vel.x + pos.y * vel.y);
+        const Scalar c = pos.x * pos.x + pos.y * pos.y - Rsq;
+        dt = (-b + sign * slow::sqrt(b * b - Scalar(4) * vxy_sq * c)) / (Scalar(2) * vxy_sq);
+
+        // backtrack the particle for dt to get to point of contact
+        pos -= vel * dt;
+
+        // update velocity according to boundary conditions
+        // no-slip requires reflection of the tangential components.
+        if (m_no_slip)
+            {
+            vel = -vel;
+            // if the particle hits the moving outside wall, then vel = -vel + 2V_t.
+            /*
+             * t = [-y/R, x/R, 0] is the tangential unit vector at the point of contact r, which is
+             * the particle position that has been backtracked to the surface in the previous step.
+             * Therefore, V_t = m_w * R * t = m_w * [-pos.y, pos.x, 0].
+             */
+            if (sign == -1)
+                {
+                vel.x += Scalar(2) * m_w * -pos.y;
+                vel.y += Scalar(2) * m_w * pos.x;
+                }
+            }
+        // slip requires reflection of the normal components.
+        /*
+         * n = r/R is the normal unit vector at the point of contact r, which is the particle
+         * position. Therefore, V_n = dot(v, n) * n = vel * pos * pos / Rsq.
+         */
+        else
+            {
+            vel.x -= Scalar(2) * pos.x * pos.x * vel.x / Rsq;
+            vel.y -= Scalar(2) * pos.y * pos.y * vel.y / Rsq;
+            }
+
+        return true;
+        }
+
+    //! Check if a particle is out of bounds
+    /*!
+     * \param pos Current particle position
+     * \returns True if particle is out of bounds, and false otherwise
+     */
+    HOSTDEVICE bool isOutside(const Scalar3& pos) const
+        {
+        const Scalar rsq = pos.x * pos.x + pos.y * pos.y;
+        return (rsq > m_R1_sq || rsq < m_R0_sq);
+        }
+
+    HOSTDEVICE void addToVirtualParticleVelocity(Scalar3& vel, const Scalar3& pos) const
+        {
+        const Scalar rsq = pos.x * pos.x + pos.y * pos.y;
+        if (rsq > m_R1_sq)
+            {
+            // Find the closest point on the perimeter of the outer cylinder.
+            const Scalar pos_x = slow::sqrt(m_R1_sq / rsq) * pos.x;
+            const Scalar pos_y = slow::sqrt(m_R1_sq / rsq) * pos.y;
+
+            vel.x += m_w * -pos_y;
+            vel.y += m_w * pos_x;
```

Comment:
I might be mistaken, but isn't the formula to go from angular to linear velocity:
```
v = w x r
```
In this case, we have `r = [pos.x, pos.y, pos.z]` and `w = [0, 0, m_w]`, so this would be just:
```suggestion
            vel.x -= m_w * pos.y;
            vel.y += m_w * pos.x;
```
?